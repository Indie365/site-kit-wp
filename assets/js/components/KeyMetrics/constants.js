/**
 * Key Metrics components - constants.
 *
 * Site Kit by Google, Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const KEY_METRICS_SETUP_CTA_WIDGET_SLUG = 'key-metrics-setup-cta-widget';
export const KEY_METRICS_SELECTION_PANEL_OPENED_KEY =
	'googlesitekit-key-metrics-selection-panel-opened';
export const KEY_METRICS_SELECTION_FORM = 'key-metrics-selection-form';
export const KEY_METRICS_SELECTED = 'key-metrics-selected';
export const MIN_SELECTED_METRICS_COUNT = 2;
export const MAX_SELECTED_METRICS_COUNT = 4;

export const KEY_METRICS_GROUP_VISITORS = {
	SLUG: 'visitors',
	LABEL: 'Visitors',
};
export const KEY_METRICS_GROUP_DRIVING_TRAFFIC = {
	SLUG: 'drivingTraffic',
	LABEL: 'Driving traffic',
};
export const KEY_METRICS_GROUP_GENERATING_LEADS = {
	SLUG: 'generatingLeads',
	LABEL: 'Generating leads',
};
export const KEY_METRICS_GROUP_SELLING_PRODUCTS = {
	SLUG: 'sellingProducts',
	LABEL: 'Selling products',
};
export const KEY_METRICS_GROUP_CONTENT_PERFORMANCE = {
	SLUG: 'contentPerformance',
	LABEL: 'Content performance',
};
