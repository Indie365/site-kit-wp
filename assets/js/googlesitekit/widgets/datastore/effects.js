/**
 * `core/effects` data store: widgets info.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import invariant from 'invariant';

/**
 * Internal dependencies
 */
import { createReducer } from '../../../../js/googlesitekit/data/create-reducer';

const REGISTER_EFFECT = 'REGISTER_EFFECT';

export const initialState = {
	effects: {},
};

export const actions = {
	/**
	 * Registers an effect with a given slug and callback.
	 *
	 * @since n.e.x.t
	 *
	 * @param {string}   slug   Widget's slug.
	 * @param {Function} effect Callback hook.
	 * @return {Object} Redux-style action.
	 */
	registerModuleEffect( slug, effect = {} ) {
		invariant( effect, 'effect is required.' );

		return {
			payload: {
				slug,
				effect,
			},
			type: REGISTER_EFFECT,
		};
	},
};

export const controls = {};

export const reducer = createReducer( ( state, { type, payload } ) => {
	switch ( type ) {
		case REGISTER_EFFECT: {
			const { slug, effect } = payload;

			if ( state.effects[ slug ] !== undefined ) {
				global.console.warn(
					`Could not register effect with slug "${ slug }". Effect "${ slug }" is already registered.`
				);

				return state;
			}

			state.effects[ slug ] = { effect, slug };
			return state;
		}

		default: {
			return state;
		}
	}
} );

export const resolvers = {};

export const selectors = {
	/**
	 * Returns all registered effects.
	 *
	 * @since n.e.x.t
	 *
	 * @param {Object} state Data store's state.
	 * @return {Array} An ordered array of widgets for this area.
	 */
	getEffects( state ) {
		return Object.values( state.effects );
	},
};

export default {
	initialState,
	actions,
	controls,
	reducer,
	resolvers,
	selectors,
};
