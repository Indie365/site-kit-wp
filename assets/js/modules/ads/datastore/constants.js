/**
 * `modules/ads` data store constants.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const MODULES_ADS = 'modules/ads';

export const ADWORDS_SCOPE = 'https://www.googleapis.com/auth/adwords';

export const PAX_SETUP_STEP = {
	LAUNCH: 1,
	FINISHED: 2,
};

// Date range offset days for Ads report requests.
export const DATE_RANGE_OFFSET = 1;
