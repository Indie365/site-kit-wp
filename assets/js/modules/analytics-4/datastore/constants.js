/**
 * `modules/analytics-4` data store constants.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

export const MODULES_ANALYTICS_4 = 'modules/analytics-4';

export const PROPERTY_CREATE = 'property_create';
export const WEBDATASTREAM_CREATE = 'webdatastream_create';

export const MAX_WEBDATASTREAMS_PER_BATCH = 10;

// Date range offset days for Analytics 4 report requests.
export const DATE_RANGE_OFFSET = 1;

export const GTM_SCOPE = 'https://www.googleapis.com/auth/tagmanager.readonly';

export const ENHANCED_MEASUREMENT_FORM = 'enhanced-measurement-form';
export const ENHANCED_MEASUREMENT_ENABLED = 'enhanced-measurement-enabled';
export const ENHANCED_MEASUREMENT_SHOULD_DISMISS_ACTIVATION_BANNER =
	'enhanced-measurement-should-dismiss-activation-banner';

export const FORM_CUSTOM_DIMENSIONS_CREATE = 'analytics4CustomDimensionsCreate';

export const CUSTOM_DIMENSION_DEFINITIONS = {
	googlesitekit_post_date: {
		parameterName: 'googlesitekit_post_date',
		displayName: __( 'WordPress Post Creation Date', 'google-site-kit' ),
		description: __(
			'Date of which this post was published',
			'google-site-kit'
		),
		scope: 'EVENT',
	},
	googlesitekit_post_author: {
		parameterName: 'googlesitekit_post_author',
		displayName: __( 'WordPress Post Author', 'google-site-kit' ),
		description: __(
			'User ID of the author for this post',
			'google-site-kit'
		),
		scope: 'EVENT',
	},
	googlesitekit_post_categories: {
		parameterName: 'googlesitekit_post_categories',
		displayName: __( 'WordPress Post Categories', 'google-site-kit' ),
		description: __(
			'Comma-separated list of category IDs assigned to this post',
			'google-site-kit'
		),
		scope: 'EVENT',
	},
	googlesitekit_post_type: {
		parameterName: 'googlesitekit_post_type',
		displayName: __( 'WordPress Post Type', 'google-site-kit' ),
		description: __( 'Content type for this post', 'google-site-kit' ),
		scope: 'EVENT',
	},
};
