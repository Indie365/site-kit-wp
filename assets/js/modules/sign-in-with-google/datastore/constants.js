/**
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const MODULES_SIGN_IN_WITH_GOOGLE = 'modules/sign-in-with-google';

export const SIGN_IN_WITH_GOOGLE_THEME_LIGHT = 'light';
export const SIGN_IN_WITH_GOOGLE_THEME_NEUTRAL = 'neutral';
export const SIGN_IN_WITH_GOOGLE_THEME_DARK = 'dark';
export const SIGN_IN_WITH_GOOGLE_THEMES = [
	SIGN_IN_WITH_GOOGLE_THEME_LIGHT,
	SIGN_IN_WITH_GOOGLE_THEME_NEUTRAL,
	SIGN_IN_WITH_GOOGLE_THEME_DARK,
];

export const SIGN_IN_WITH_GOOGLE_TEXT_CONTINUE_WITH_GOOGLE =
	'Continue with Google';
export const SIGN_IN_WITH_GOOGLE_TEXT_SIGN_IN = 'Sign in';
export const SIGN_IN_WITH_GOOGLE_TEXT_SIGN_IN_WITH_GOOGLE =
	'Sign in with Google';
export const SIGN_IN_WITH_GOOGLE_TEXT_SIGN_UP_WITH_GOOGLE =
	'Sign up with Google';
export const SIGN_IN_WITH_GOOGLE_TEXTS = [
	SIGN_IN_WITH_GOOGLE_TEXT_CONTINUE_WITH_GOOGLE,
	SIGN_IN_WITH_GOOGLE_TEXT_SIGN_IN,
	SIGN_IN_WITH_GOOGLE_TEXT_SIGN_IN_WITH_GOOGLE,
	SIGN_IN_WITH_GOOGLE_TEXT_SIGN_UP_WITH_GOOGLE,
];

export const SIGN_IN_WITH_GOOGLE_SHAPE_PILL = 'pill';
export const SIGN_IN_WITH_GOOGLE_SHAPE_RECTANGULAR = 'rectangular';
export const SIGN_IN_WITH_GOOGLE_SHAPES = [
	SIGN_IN_WITH_GOOGLE_SHAPE_PILL,
	SIGN_IN_WITH_GOOGLE_SHAPE_RECTANGULAR,
];
