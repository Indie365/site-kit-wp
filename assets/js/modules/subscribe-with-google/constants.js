/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

export const REVENUE_MODELS = [
	{
		displayName: __( 'Contributions', 'google-site-kit' ),
		value: 'contribution',
	},
	{
		displayName: __( 'Subscriptions', 'google-site-kit' ),
		value: 'subscription',
	},
];
