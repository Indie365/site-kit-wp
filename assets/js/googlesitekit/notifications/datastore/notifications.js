/**
 * `core/notifications` data store: notifications info.
 *
 * Site Kit by Google, Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import Data from 'googlesitekit-data';
import { createReducer } from '../../../../js/googlesitekit/data/create-reducer';
import { CORE_NOTIFICATIONS } from './constants';
import { CORE_USER } from '../../datastore/user/constants';

const { createRegistrySelector } = Data;

const REGISTER_NOTIFICATION = 'REGISTER_NOTIFICATION';
const RECEIVE_ACTIVE_IDS = 'RECEIVE_ACTIVE_IDS';

export const initialState = {
	notifications: {},
	activeIDs: undefined,
};

export const actions = {
	registerNotification(
		id,
		{ type, shouldDisplay, Component, dismissable }
	) {
		return {
			payload: {
				id,
				settings: {
					type,
					shouldDisplay,
					Component,
					dismissable,
				},
			},
			type: REGISTER_NOTIFICATION,
		};
	},

	receiveActiveIDs( activeIDs ) {
		return {
			payload: {
				activeIDs,
			},
			type: RECEIVE_ACTIVE_IDS,
		};
	},
};

export const controls = {};

export const reducer = createReducer( ( state, { type, payload } ) => {
	switch ( type ) {
		case REGISTER_NOTIFICATION: {
			const { id, settings } = payload;

			if ( state.notifications[ id ] !== undefined ) {
				global.console.warn(
					`Could not register notification with ID "${ id }". Notification "${ id }" is already registered.`
				);
			} else {
				state.notifications[ id ] = { ...settings, id };
			}

			break;
		}

		case RECEIVE_ACTIVE_IDS: {
			state.activeIDs = payload.activeIDs;
			break;
		}

		default:
			break;
	}
} );

export const resolvers = {
	*getActiveNotifications() {
		const registry = yield Data.commonActions.getRegistry();

		const notifications = registry
			.select( CORE_NOTIFICATIONS )
			.getNotifications();

		const resolved = yield Data.commonActions.await(
			Promise.all(
				notifications.map( ( notification ) =>
					notification.shouldDisplay( registry )
				)
			)
		);

		const activeIDs = [];
		notifications.forEach( ( notification, idx ) => {
			if ( resolved[ idx ] ) {
				activeIDs.push( notification.id );
			}
		} );

		yield actions.receiveActiveIDs( activeIDs );
	},
};

export const selectors = {
	getNotifications: createRegistrySelector( ( select ) => ( state ) => {
		const order = [ 'error', 'warning', 'info' ];

		return Object.values( state.notifications )
			.filter( ( notification ) => {
				if (
					notification.dismissable &&
					select( CORE_USER ).isItemDismissed( notification.id )
				) {
					return false;
				}

				return true;
			} )
			.sort( ( a, b ) => {
				return order.indexOf( a.type ) - order.indexOf( b.type );
			} );
	} ),

	getActiveNotifications: ( state ) => {
		if ( state.activeIDs === undefined ) {
			return undefined;
		}

		return selectors
			.getNotifications( state )
			.filter( ( notification ) =>
				state.activeIDs.includes( notification.id )
			);
	},
};

export default {
	initialState,
	actions,
	controls,
	reducer,
	resolvers,
	selectors,
};
