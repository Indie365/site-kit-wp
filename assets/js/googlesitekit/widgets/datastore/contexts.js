/**
 * `core/widgets` data store: contexts.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import invariant from 'invariant';

/**
 * Internal dependencies
 */
import Data from 'googlesitekit-data';
import { CORE_WIDGETS } from './constants';

const { createRegistrySelector } = Data;

export const selectors = {
	/**
	 * Checks if a widget context is active.
	 *
	 * Returns `true` if the widget context is active.
	 * Returns `false` if the widget context is NOT active.
	 *
	 * @since n.e.x.t
	 *
	 * @param {Object} state Data store's state.
	 * @param {string} slug  Widget context's slug.
	 * @return {boolean} `true`/`false` based on whether widget context is active.
	 */
	isWidgetContextActive: createRegistrySelector(
		( select ) => ( state, contextSlug ) => {
			invariant(
				contextSlug,
				'contextSlug is required to check a widget context is active.'
			);

			return select( CORE_WIDGETS )
				.getWidgetAreas( contextSlug )
				.some( ( area ) =>
					select( CORE_WIDGETS ).isWidgetAreaActive( area.slug )
				);
		}
	),
};

export default {
	selectors,
};
