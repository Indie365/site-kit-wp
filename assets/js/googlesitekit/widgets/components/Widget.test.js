/**
 * Widget component tests.
 *
 * Site Kit by Google, Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import Widget from './WidgetRenderer';
import { STORE_NAME } from '../datastore/constants';
import { render } from '../../../../../tests/js/test-utils';

const setupRegistry = ( { component = () => <div>Test</div>, dispatch } ) => {
	dispatch( STORE_NAME ).registerWidgetArea( 'dashboard-header', {
		title: 'Dashboard Header',
		subtitle: 'Cool stuff for yoursite.com',
		style: 'boxes',
	} );
	dispatch( STORE_NAME ).assignWidgetArea( 'dashboard-header', 'dashboard' );
	dispatch( STORE_NAME ).registerWidget( 'TestWidget', {
		component,
	} );
	dispatch( STORE_NAME ).assignWidget( 'TestWidget', 'dashboard-header' );
};

describe( 'Widget', () => {
	it( 'should output children directly', async () => {
		const { container } = render( <Widget slug="TestWidget" />, { setupRegistry } );

		expect( Object.values( container.firstChild.classList ) ).toEqual( [] );
		expect( container.firstChild ).toMatchSnapshot();
	} );

	it( 'should output null when no slug is found', async () => {
		const { container } = render( <Widget slug="NotFound" />, { setupRegistry } );

		expect( container.firstChild ).toEqual( null );
	} );
} );
