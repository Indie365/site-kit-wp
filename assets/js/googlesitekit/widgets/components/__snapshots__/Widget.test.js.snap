// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget should output children directly 1`] = `
<div>
  Test
</div>
`;
